flows/inference.py

```diff
@@ -116,6 +118,69 @@ def to_json(self) -> dict:
         }
 
 
+class BatchInferenceException(Exception):
+    """
+    Exception raised when batch inference fails.
+
+    The data attribute of the prefect StateType.FAILED must be an exception. Thus, we
+    declare a custom exception that can also be used to transfer the results of the
+    batch inference.
+    """
+
+    def __init__(self, message: str, data: dict[str, Any]):
+        self.message = message
+        self.data = data
+
+
+class BatchInferenceResult(BaseModel):
+    """Result from running inference on a batch of documents."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    successful_document_stems: set[DocumentStem]
+    failed_document_stems: set[tuple[DocumentStem, Exception]]
+    classifier_name: str
+    classifier_alias: str
+
+    @property
+    def failed(self) -> bool:
+        """Whether the batch failed, True if failed."""
+
+        return self.failed_document_stems != set()
```

Comment:
Do we need this, can't we just get the State object from the deployment result and then use that to decide whether the batch failed or not rather than an inferred value.